src/main/java/net/neoforged/neoforge/common/conditions/FlagCondition.java

```diff
@@ -5,68 +5,53 @@
 
 package net.neoforged.neoforge.common.conditions;
 
-import com.mojang.serialization.Codec;
 import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import net.minecraft.world.flag.FeatureFlag;
 import net.minecraft.world.flag.FeatureFlagSet;
 import net.minecraft.world.flag.FeatureFlags;
+import org.apache.commons.lang3.ArrayUtils;
 
 /**
- * Condition checking for the enabled state of a given {@link FeatureFlagSet}.
- * <p>
- * {@code requiredFeatures} - {@link FeatureFlagSet} containing all {@link FeatureFlag feature flags} to be validated.
- * {@code expectedResult} - Validates that all given {@link FeatureFlag feature flags} are enabled when {@code true} or disabled when {@code false}.
+ * Condition checking that a set of {@link FeatureFlag feature flags} are enabled.
  *
  * @apiNote Mainly to be used when flagged content is not contained within the same feature pack which also enables said {@link FeatureFlag feature flags}.
  */
 public final class FlagCondition implements ICondition {
     public static final MapCodec<FlagCondition> CODEC = RecordCodecBuilder.mapCodec(instance -> instance.group(
-            FeatureFlags.CODEC.fieldOf("flags").forGetter(condition -> condition.requiredFeatures),
-            Codec.BOOL.lenientOptionalFieldOf("expected_result", true).forGetter(condition -> condition.expectedResult)).apply(instance, FlagCondition::new));
+            FeatureFlags.CODEC.fieldOf("flags").forGetter(condition -> condition.requiredFeatures)).apply(instance, FlagCondition::new));
 
     private final FeatureFlagSet requiredFeatures;
-    private final boolean expectedResult;
 
-    private FlagCondition(FeatureFlagSet requiredFeatures, boolean expectedResult) {
+    private FlagCondition(FeatureFlagSet requiredFeatures) {
+        if (requiredFeatures.isEmpty()) {
+            throw new IllegalArgumentException("FlagCondition requires a non-empty feature flag set");
+        }
         this.requiredFeatures = requiredFeatures;
-        this.expectedResult = expectedResult;
     }
 
     @Override
     public boolean test(IContext context) {
-        var flagsEnabled = requiredFeatures.isSubsetOf(context.enabledFeatures());
-        // true if: 'expectedResult' is true nd all given flags are enabled
-        // false if: `enabledEnabled' is false and all given flags are disabled
-        return flagsEnabled == expectedResult;
+        return requiredFeatures.isSubsetOf(context.enabledFeatures());
     }
 
     @Override
     public MapCodec<? extends ICondition> codec() {
         return CODEC;
     }
 
-    public static ICondition isEnabled(FeatureFlagSet requiredFeatures) {
-        return new FlagCondition(requiredFeatures, true);
+    public static ICondition of(FeatureFlagSet requiredFeatures) {
```

Comment:
Not a fan of this name, as by simply reading it you are unsure if it checks for given flags being enabled or disabled.

If we must have method be called `of` and are dropping the `isDisabled` functionality, maybe we rename the class to being `FlagsEnabledCondition` instead, this would also following the naming convention of other conditions like `ModLoadedCondition`